src/main/java/com/fr3ts0n/androbd/plugin/mgr/PluginHandler.java

```diff
@@ -155,21 +158,24 @@ public void onServiceDisconnected(ComponentName name)
 
     /**
      * Adds or replaces a single item in this ArrayAdapter
-     * @param item
+     * @param item The item to insert or replace
+     * @return true if the item was replaced, false if it was added as a new item
      */
-    public void upsert(PluginInfo item)
+    public boolean upsert(PluginInfo item)
     {
         int index = getPosition(item);
         if (index < 0)
         {
             add(item);
+            return false;
         }
         else
         {
             setNotifyOnChange(false);
```

Comment:
Why suppress notification of the removal, but notify the Addition? 
I would suspect to either notify both, or None in Case of replacement.

Reply:
I assume that the ArrayAdapter has `setNotifyOnChange` True by default, and so I don't modify this flag during the single operation of the addition flow. When replacing an item, I toggle it off for the delete and turn it back on for the addition, so it only notifies any ArrayAdapter listeners one single time for these two operations.
I felt it was important to continue to notify for the replacement, just in case the new PluginInfo has different flags or other difference.

Reply:
I would expect the notifications to be different for insertion and removal. Therefore the visual list would only add a element without removing the old one ...
So, to be on the clean side I would either notify on both events, or none if visual update is not needed / expected.

Reply:
You are right, it adds too much complication to be worth the tiny optimization, and so I removed it.